xbmc/playlists/SmartPlayList.cpp

```diff
@@ -262,14 +265,14 @@ bool CSmartPlaylistRule::Validate(const std::string &input, void *data)
     return true;
 
   // split the input into multiple values and validate every value separately
-  std::vector<std::string> values = StringUtils::Split(input, RULE_VALUE_SEPARATOR);
-  for (std::vector<std::string>::const_iterator it = values.begin(); it != values.end(); ++it)
-  {
-    if (!validator(*it, data))
-      return false;
-  }
+  std::vector<std::string> values;
+  if (input.empty())
+    values = {""};
+  else
+    values = StringUtils::Split(input, RULE_VALUE_SEPARATOR);
```

Comment:
I don't think we need to handle empty input special. `StringUtils::Split` won't insert anything into values and `std::all_of` on an empty range is `true`.

Reply:
It's an empty element on purpose to run the validation on empty input, the objective of that commit.
I can add a comment to avoid someone removing later by mistake.

Reply:
Actually I'll call validator explicitly with "" for empty input and there will be no confusion possible ever and no comment needed.